### PR TITLE
Stubs for humanfriendly

### PR DESCRIPTION
**Pull request by ismeta**
Autogenerated with stubgen + manual adjustments to fix e.g. namedtuple, Python versioning, removing unneeded imports, required stubtest packages/allowlist.

cc @JukkaL

### CODE REVIEW

File: stubs/humanfriendly/humanfriendly/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+class InvalidDate(Exception): ...
+class InvalidSize(Exception): ...
+class InvalidLength(Exception): ...
+class InvalidTimespan(Exception): ...
+
+# Names in __all__ with no definition:
+#   CombinedUnit
+#   SizeUnit
+#   __version__
```
Comment (JukkaL) [inline stubs/humanfriendly/humanfriendly/__init__.pyi:53]: This comment can be removed.

File: stubs/humanfriendly/humanfriendly/tests.pyi [hunk 2]
```diff
@@ -0,0 +1,82 @@
+from typing import Any
```
Comment (JukkaL) [inline stubs/humanfriendly/humanfriendly/tests.pyi:1]: This file can be removed since it only contains tests.

**Review by JukkaL — COMMENTED** [review]
Thanks for the PR! Looks good, I just found a few things that can probably be removed.

File: stubs/humanfriendly/humanfriendly/case.pyi [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+class CaseInsensitiveDict(collections.OrderedDict):
+    def __init__(self, other: Any | None = ..., **kw) -> None: ...
+    def coerce_key(self, key): ...
+    @classmethod
+    def fromkeys(cls, iterable, value: Any | None = ...): ...
+    def get(self, key, default: Any | None = ...): ...
+    def pop(self, key, default: Any | None = ...): ...
+    def setdefault(self, key, default: Any | None = ...): ...
+    def update(self, other: Any | None = ..., **kw) -> None: ...  # type: ignore[override]
```
Comment (JelleZijlstra) [inline stubs/humanfriendly/humanfriendly/case.pyi:14]:
```suggestion
    def update(self, other: Any | None = ..., **kw) -> None: ...  # type: ignore
```
This is mypy-specific and may trip up other type checkers

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**